lib/file.cpp

```diff
@@ -53,6 +53,13 @@ file file::text(std::string_view prefix) {
     return file(prefix, /*binary=*/false);
 }
 
+file file::copy(const fs::path& path, std::string_view prefix) {
+    file tmpfile = file(prefix);
```

Comment:
Don't you need to check that it is indeed a file?

No need to copy recursively either